src/protocol/message_layer/StreamMessageV28.js

```diff
@@ -0,0 +1,75 @@
+import UnsupportedVersionError from '../../errors/UnsupportedVersionError'
+import StreamMessage from './StreamMessage'
+import StreamMessageV29 from './StreamMessageV29'
+import StreamMessageV30 from './StreamMessageV30'
+
+const BYE_KEY = '_bye'
+
+const VERSION = 28
+
+export default class StreamMessageV28 extends StreamMessage {
+    constructor(streamId, streamPartition, timestamp, ttl, offset, previousOffset, contentType, content) {
+        super(VERSION, ttl, contentType, content)
+        this.streamId = streamId
+        this.streamPartition = streamPartition
+        this.timestamp = timestamp
+        this.offset = offset
+        this.previousOffset = previousOffset
+    }
+
+    getStreamId() {
+        return this.streamId
+    }
+
+    toObject(parsedContent = false, compact = true) {
```

Comment:
Seeing that this is an older version, probably not worth changing anymore, but in general it may be worth splitting such flag options into their own methods. When reading in the context of the client code (user of this method) it becomes more apparent what they are trying to achieve.

I.e.
```
x.toObject(false, true)
x.toObject(false, false)
```
vs.
x.toObject(false)
x.toArray(false)
```